platform/emscripten/emscripten.cmake

```diff
@@ -0,0 +1,132 @@
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/file_source_request.cpp
+        # ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/http_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/local_file_request.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/local_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/mbtiles_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/main_resource_loader.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline_database.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/offline_download.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/online_file_source.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/storage/sqlite3.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/text/bidi.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/text/local_glyph_rasterizer.cpp
+        
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/compression.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/filesystem.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/image.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/jpeg_reader.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/webp_reader.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/logging_stderr.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/monotonic_timer.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/png_reader.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/png_writer.cpp
+        
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/string_stdlib.cpp
+        
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/utf.cpp
+
+        
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/thread.cpp
+        ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/util/thread_local.cpp
+        ${PROJECT_SOURCE_DIR}/platform/emscripten/src/mbgl/util/async_task.cpp
+        ${PROJECT_SOURCE_DIR}/platform/emscripten/src/mbgl/util/timer.cpp
+        
+        # ${PROJECT_SOURCE_DIR}/platform/emscripten/src/mbgl/util/logging.cpp
+
+        # GLFW
+        # ${PROJECT_SOURCE_DIR}/platform/glfw/glfw_view.cpp
+        # ${PROJECT_SOURCE_DIR}/platform/glfw/glfw_renderer_frontend.cpp
+        # ${PROJECT_SOURCE_DIR}/platform/glfw/settings_json.cpp
+        # ${PROJECT_SOURCE_DIR}/platform/default/src/mbgl/map/map_snapshotter.cpp
+
+        # ${PROJECT_SOURCE_DIR}/platform/glfw/glfw_gl_backend.cpp
+
+        ${PROJECT_SOURCE_DIR}/platform/emscripten/src/mbgl/headless_backend_glfw.cpp
+        ${PROJECT_SOURCE_DIR}/platform/windows/src/gl_functions.cpp
+)
+
+add_executable(mbgl-core ${INCLUDE_FILES} ${SRC_FILES} ${EXTRA_SRC_FILES})
```

Comment:
The way this is going... with the SIDE_MODULE shenanigans and all... it would be far easier to simply reference these 3 variables and the libraries below into standalone wasm projects.